seqr/views/apis/anvil_workspace_api.py

```diff
@@ -1,57 +1,133 @@
+        logger.warning(message)
+        return create_json_response({'error': message}, status=400, reason=message)
+
+    project_name = _get_project_name(namespace, name)
+    project = Project.objects.filter(name=project_name)
+    if project:
+        return redirect('/project/{}/project_page'.format(project.first().guid))
+    else:
+        return redirect('/create_project_from_workspace/{}/{}'.format(namespace, name))
 
 
 @login_required(login_url=API_LOGIN_REQUIRED_URL)
 def create_project_from_workspace(request, namespace, name):
     """
-    Create a project when a cooperator requesting to load data from an AnVIL workspace.
+    Create a project when a cooperator requests to load data from an AnVIL workspace.
 
-    Args:
-        request (object): Django request object
-        namespace: The namespace (or the billing account) of the workspace
-        name: The name of the workspace. It also be used as the project name
-    Return:
-        The projectsByGuid with the new project json
+    :param request: Django request object
+    :param namespace: The namespace (or the billing account) of the workspace
+    :param name: The name of the workspace. It also be used as the project name
+    :return the projectsByGuid with the new project json
 
     """
-    response_json = {}  # to be done
-    # Todo:
-    # 1) Validate that the current user has logged in through google and has one of the valid can_edit levels of
-    #  access on the specified workspace;
-    # 2) Validate all the user input from the post body;
-    # 3) Add the seqr service account to the corresponding AnVIL workspace, so that our team will have access to the
-    # project for data loading;
-    # 4) Create a new Project in seqr. This project should NOT be added to the analyst group. The project name should
-    #  just be the workspace name. Make sure to set workspace_namespace and workspace_name correctly;
-    # 5) Add families/individuals based on the uploaded pedigree file;
-    # 6) Send an email to all seqr data managers saying a new AnVIL project is ready for loading. Include the seqr
-    #  project guid, the workspace name, and attach a txt file with a list of the individual IDs that were created.
-    return create_json_response(response_json)
+    project_name = _get_project_name(namespace, name)
+    project = Project.objects.filter(name = project_name)
+    if project:
+        error = 'Project {} exists.'.format(project_name)
+        logger.warning(error)
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    # Validate that the current user has logged in through google and has sufficient permissions
+    if not workspace_has_perm(request.user, CAN_EDIT, namespace, name, can_share=True):
+        error = "Missing required permissions for loading data from workspace {}/{}".format(namespace, name)
+        logger.warning(error)
+        return create_json_response({'error': error}, status=400, reason=error)
```

Comment:
unauthorized requests should raise a `PermissionDenied` error (which will be correctly logged and handled downstream as a 403 response), not return a 400 error. I think you would be better off here if instead of using the `workspace_has_perm` helper you created a `check_workspace_permissions` function similar to `check_project_permissions` that will raise the exception for you and then you don't need to handle it in the code here